### PR TITLE
Added features and QoL improvements.

### PR DESCRIPTION
**Pull request by wrefgtzweve**
Fixed spectators not seeing themselves causing confusion.
Added vgui panel for settings
Replaced the hitbox logic with chams / xray

image
image
image

### CODE REVIEW

File: lua/fspectate/cl_init.lua [hunk 1]
```diff
@@ -512,6 +514,22 @@ local function drawHelp()
     draw.WordBox( 2, bottomLeft.x, bottomLeft.y + 12, "Left click to spectate!", "UiBold", uiBackground, uiForeground )
 end
 
+--[[---------------------------------------------------------------------------
+Draws a crosshair
+---------------------------------------------------------------------------]]
+local function drawCrosshair()
+    if not showCrosshair then return end
+    local h = ScrH() / 2
+    local w = ScrW() / 2 - 1 -- -1 to make it exactly match with the hl2 crosshair.
```
Comment (plally) [inline lua/fspectate/cl_init.lua:570]: 🤨

File: lua/fspectate/cl_init.lua [hunk 2]
```diff
@@ -512,6 +514,22 @@ local function drawHelp()
...
+
+    surface.SetDrawColor( 0, 0, 0, 255 )
+    surface.DrawLine( w + 1, h + 10, w + 1, h - 10 )
+    surface.DrawLine( w + 10, h + 1, w - 10, h + 1 )
+    surface.SetDrawColor( 255, 255, 255, 255 )
```
Comment (plally) [inline lua/fspectate/cl_init.lua]: would personally prefer green with black outline.

File: lua/fspectate/cl_init.lua (hunk 1 above)
Comment (wrefgtzweve) [reply lua/fspectate/cl_init.lua:570]: What's the issue? try it for yourself, it really does need the -1 due to sorse.

File: lua/fspectate/cl_init.lua (hunk 2 above)
Comment (wrefgtzweve) [reply lua/fspectate/cl_init.lua]: Was planning a follow up PR with even more customizability, could add a custom color option if you want. If you make a quick suggestion for the color change i'll add that.

File: lua/fspectate/cl_init.lua [hunk 3]
```diff
@@ -135,50 +208,30 @@ local function specCalcView()
...
+    local col = team.GetColor( ply:Team() )
+    cam.Start3D();
+        render.SetColorModulation( col.r / 1000, col.g / 1000, col.b / 1000 )
+        render.MaterialOverride( chamsmat1 )
+        ply:DrawModel()
+        render.SetColorModulation( col.r / 255, col.g / 255, col.b / 255 )
+        render.MaterialOverride( chamsmat2 )
+        ply:DrawModel()
+        render.MaterialOverride()
+    cam.End3D();
```
Comment (sarahsturgeon) [inline lua/fspectate/cl_init.lua]: Given how frequently this runs and how important framerate is while recording, I think it'd be nice to get any savings we can.

```suggestion
local chamsmat2 = CreateMaterial( "CHAMSMATFSPEC2", "VertexLitGeneric", {["$basetexture"] = "models/debug/debugwhite", ["$model"] = 1, ["$ignorez"] = 0} )

local teamGetColor = team.GetColor
local Start3D = cam.Start3D
local SetColorModulation = render.SetColorModulation
local MaterialOverride = render.MaterialOverride
local End3D = cam.End3D

local function drawCham( ply )
    if not ply:Alive() then return end
    if ply == specEnt and not thirdperson then return end

    local col = ply._teamColor
    if not col then
        col = teamGetColor( ply:Team() )
        ply._teamColor = col
    end

    local r, g, b = col:Unpack()

    Start3D();
        SetColorModulation( r / 1000, g / 1000, b / 1000 )
        MaterialOverride( chamsmat1 )

        ply:DrawModel()

        SetColorModulation( r / 255, g / 255, b / 255 )
        MaterialOverride( chamsmat2 )

        ply:DrawModel()
        MaterialOverride()
    End3D();
```

File: lua/fspectate/cl_init.lua [hunk 4]
```diff
@@ -400,20 +451,19 @@ local function drawHelp()
...
-            draw.WordBox( 2, 101, scrHalfH + 100, specEnt:Nick() .. " " .. specEnt:SteamID(), "UiBold", uiBackground, team.GetColor( specEnt:Team() ) )
+            draw.WordBox( 2, 10, scrHalfH + 80, "Spectating: ", "UiBold", uiBackground, uiForeground )
+            draw.WordBox( 2, 101, scrHalfH + 80, specEnt:Nick() .. " " .. specEnt:SteamID(), "UiBold", uiBackground, team.GetColor( specEnt:Team() ) )
         else
-            draw.WordBox( 2, 10, scrHalfH + 100, "Owner: ", "UiBold", uiBackground, uiForeground )
+            draw.WordBox( 2, 10, scrHalfH + 80, "Owner: ", "UiBold", uiBackground, uiForeground )
 
             if specEnt:CPPIGetOwner() then
-                draw.WordBox( 2, 70, scrHalfH + 100, specEnt:CPPIGetOwner():Nick() .. " " .. specEnt:CPPIGetOwner():SteamID(), "UiBold", uiBackground, team.GetColor( specEnt:CPPIGetOwner():Team() ) )
+                draw.WordBox( 2, 70, scrHalfH + 80, specEnt:CPPIGetOwner():Nick() .. " " .. specEnt:CPPIGetOwner():SteamID(), "UiBold", uiBackground, team.GetColor( specEnt:CPPIGetOwner():Team() ) )
```
Comment (sarahsturgeon) [inline lua/fspectate/cl_init.lua]: CPPIGetOwner is a little heavy, might be nice to store the name, steamid, and team(team color?) in a variable that changes when the spectator changes

**Review by sarahsturgeon — COMMENTED** [review]
Just a couple of performance-driven comments

**Pull request merged**